### PR TITLE
ICMP security groups documentation

### PR DESCRIPTION
**Pull request by hyprbst**
This adds a section on letting servers ping eachother through the security groups mechanism. It is a commonly asked question regarding the `code` and `type` numbers required for pinging servers.

### CODE REVIEW

**Comment by swat30** [comment]
Thanks @hyprbst ! Looks good

File: content/networking/security-groups.md [hunk 1]
```diff
@@ -68,3 +68,15 @@ column of assigned groups.
...
+## ICMP / Ping Rules
+
+In order to allow servers to ping each other, or be reached via ICMP through
+the internet, you must create a Security Group Rule of protocol **ICMP**, type
+**-1**, and code **-1**.
+
+![Security Group Rule ICMP](/img/content/networking/security-group-icmp.png)
+
+The above screenshot shows an example of opening ICMP traffic to all internal
+traffic on the `10.0.0.0/24` subnet, which will only allow the traffic for
```
Comment (thurloat) [inline content/networking/security-groups.md]: this sentence feels awkward -- perhaps "traffic to all internal servers"?

Comment (swat30) [inline content/networking/security-groups.md]: agreed, it should probably be more generic. Although the 10.0.0.0/24 subnet is good to have as an example

Comment (hyprbst) [inline content/networking/security-groups.md]: Sorry about that! will get this fixed and pushed again asap.

:cake:

**Comment by hyprbst** [comment]
@swat30 do you want the CIDR example to be separated into it's own section, and leave the ICMP section to just it's own information?

**Comment by swat30** [comment]
Good point @hyprbst. Maybe we should split all examples out into a separate doc / separate docs and have a more detailed explanation of the components of security groups and rules in the main doc

**Comment by thurloat** [comment]
That sounds about right to me @swat30, if I understand correctly -- you're looking for:
- main security groups page with details on how they work / what it all means
- separate page with common examples and use cases.

I don't think we should splinter out for each example necessarily, I think the nav could get pretty cluttered. The framework automatically creates anchors for headings, maybe we could generate a page subnav, or would at least be easy to permalink to certain sections through direct link.

**Comment by swat30** [comment]
I like that idea @thurloat, pretty much what I was going for. @hyprbst do you want to update your PR to reflect these changes so we can approve?

**Comment by hyprbst** [comment]
I'll go back to the drawing board with this, thanks for the guidance. :smile:

**Comment by hyprbst** [comment]
@swat30 @thurloat can you have another look?

File: content/networking/security-groups.md [hunk 2]
```diff
@@ -1,70 +1,43 @@
...
-can belong to multiple security groups, so for a web server, you might want
-ports 22, 443, and 80 open. Or for a DB server you might want port 5432 open
-for PostgreSQL.
+Network Security Groups allow you to open traffic on a port-by-port basis. 
+This flexible model can extend to create any number of situations, whether
+simple or complex while maintaining strict security.
 
-If access is attempted to a port/IP combination that is not assigned, no
-connection will be established at all and the connection will time out.
+## Restricted by Default
```
Comment (swat30) [inline content/networking/security-groups.md]: Let's bring these headers down to h3. They look a little big for the content that follows

File: content/networking/security-groups.md [hunk 3]
```diff
@@ -1,70 +1,43 @@
...
 
-The default security group has just port 22 open, so we'll create a brand new
-one named "Web" for a web server as an example.
+By default, a raw instance only has port 22 open so you can SSH in after 
+it's created. We don't want to expose your servers to anything a brand new
```
Comment (swat30) [inline content/networking/security-groups.md]: Maybe elaborate a little bit on the security implications a little further. "Exposing all ports, or ports which don't need to be exposed, can lead to unauthorized access, etc."

File: content/networking/security-groups.md [hunk 4]
```diff
@@ -1,70 +1,43 @@
...
+The best practise for creating your security groups is to make them as 
+**specific** to a use case as possible in order to retain flexibility. The
+screenshot below shows an example of use-case groups being assigned to a server.
 
-Then, edit your security group's rules to add the required ports by clicking
-"Edit Rules".
+![Security Group Assignment](/img/content/networking/security-group-multi-assign.png)
 
-![Web Security Group](/img/content/networking/security-group-list.png)
+For example, one might create individual Security groups for SSH access, Web
```
Comment (swat30) [inline content/networking/security-groups.md]: "Security Groups" or "security groups"

Comment (swat30) [inline content/networking/security-groups.md]: caps are inconsistent in general here

File: content/networking/security-groups.md [hunk 5]
```diff
@@ -1,70 +1,43 @@
...
+screenshot below shows an example of use-case groups being assigned to a server.
 
-Then, edit your security group's rules to add the required ports by clicking
-"Edit Rules".
+![Security Group Assignment](/img/content/networking/security-group-multi-assign.png)
 
-![Web Security Group](/img/content/networking/security-group-list.png)
+For example, one might create individual Security groups for SSH access, Web 
+Server ports, DB server ports, and VPN access. A single instance (even 
+individual ports on the same server) can belong to multiple security groups.
```
Comment (swat30) [inline content/networking/security-groups.md]: Nitpick: instances don't "belong" to security groups, security group rules do :). I would use "associated" instead

File: content/networking/security-groups.md [hunk 6]
```diff
@@ -1,70 +1,43 @@
...
 
-Then, edit your security group's rules to add the required ports by clicking
-"Edit Rules".
+![Security Group Assignment](/img/content/networking/security-group-multi-assign.png)
 
-![Web Security Group](/img/content/networking/security-group-list.png)
+For example, one might create individual Security groups for SSH access, Web 
+Server ports, DB server ports, and VPN access. A single instance (even 
+individual ports on the same server) can belong to multiple security groups. 
+For an application server, you could assign the server to all of the above
```
Comment (swat30) [inline content/networking/security-groups.md]: I think assign security groups -> instance makes more sense

**Comment by swat30** [comment]
I think that your "edit instance" snap should include an unassociated security group to show what each looks like

File: content/networking/security-groups-examples.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+network used the mask `255.255.0.0` for a larger address space, you would
+create a CIDR of `10.0.0.0/16`.
+
+### Single Address
+
+Finally, if you want to only allow access for the rules of a security group to
+a single IP (office / home), you would create the most restrictive CIDR 
+possible, allowing only 1 address through using a `/32` designation. For 
+example, if your address were `76.11.212.12`, the CIDR would be 
+`76.11.212.12/32`.
```
Comment (swat30) [inline content/networking/security-groups-examples.md]: May be worth noting that you can leave the `/32` off of the IP address as well

**Comment by hyprbst** [comment]
@swat30 the screenshot shows a list of both associated and unassociated groups, or are you suggesting that I include a screenshot of the instance detail screen that lists the security groups?

**Comment by thurloat** [comment]
LGTM :+1:

**Comment by swat30** [comment]
Cool all good :+1:

**Pull request merged**